department-of-veterans-affairs/vets-website
# Add analytics to direct deposit

## Summary

- Adds new GA events for the api requests for new direct deposit endpoint `profile/direct_deposits/disability_compensations` 
- The solution was to add analytics methods to the `DirectDepositClient` class and when requests are made, then the class can call the right GA events based on lighthouse vs legacy ppiu endpoint usage.
  - One of the not so fun parts was making sure that the legacy analytics events stayed consistent, but luckily the unit tests kept me honest making sure that the event names stayed correct. From what I can tell, the old events were set up a long time ago before we had standardization of api_call events. Once we fully migrate to lighthouse for this endpoint, then there will be a good amount of code complexity that will be able to be deleted.
- This is for Profile team and we will own maintenance
- The lighthouse direct deposit work uses `profile_lighthouse_direct_deposit` that is already established as a toggle from previous work.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/54893

## Testing done

- Updated tests to actions that fire analytics events on legacy endpoint
- Added unit tests for DirectDepositClient methods

- Testing locally:
  - Run mock server and profile FE in watch mode
  - Load up profile / direct deposit page
  - The feature toggle is turned on in the mock server, so it will make a call to the lighthouse based endpoint, and will receive the a successful mocked GET response.
  - Open the chrome devtools and inspect the window.dataLayer array to view the analytics events that _would_ be fired when on staging/prod. You will see the newly added `api_call` events with `GET /profile/direct_deposits/disability_compensations`. There will be 2 events in the array, one for started and one for successful.

![Screenshot 2023-06-06 at 8 22 13 AM](https://github.com/department-of-veterans-affairs/vets-website/assets/8332986/c4ff4023-795e-4479-ae2b-5d758d68cf57)

## Screenshots

No UI changes, just analytics events work.

## What areas of the site does it impact?

Profile - Direct Deposit

## Acceptance criteria

- [x] Analytics events added for new endpoint calls

### Quality Assurance & Testing


### Error Handling
